packages/devops/scripts/ci/triggerRedeploy.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+if [[ $CI_BRANCH == "master" ]]; then
+    DEPLOYMENT_URL="tupaia.org"
+else
+    DEPLOYMENT_URL="${CI_BRANCH}.tupaia.org"
+fi
+EXISTING_INSTANCES=$(aws ec2 describe-instances \
+      --filters Name=tag:Branch,Values=${CI_BRANCH} Name=tag-key,Values=SubdomainsViaGateway Name=instance-state-name,Values=running,stopped \
+      --no-cli-pager)
+
+if [[ $EXISTING_INSTANCES == *"Instances"* ]]; then
+  echo "Existing deployment, triggering redeploy"
+  RESPONSE_FILE=lambda_redeploy_response.json
+  aws lambda invoke \
+    --function-name deployment \
+    --payload "{\"Action\": \"redeploy_tupaia_server\", \"Branch\": \"$CI_BRANCH\" }" \
+    --cli-binary-format raw-in-base64-out $RESPONSE_FILE \
+    --no-cli-pager
+
+  if grep -q errorMessage "$RESPONSE_FILE"; then
+    echo "Error while trying to redeploy"
+    cat $RESPONSE_FILE
+    exit 1
+  fi
+
+  NEW_INSTANCE_ID=$(cat $RESPONSE_FILE | cut -d "\"" -f 2)
+
+  echo "Waiting for instance to run its startup build script. To watch detailed progress, connect to instance ${NEW_INSTANCE_ID} and run tail -f logs/deployment_log.txt"
+  WAIT_ATTEMPTS=0
+  while true; do
+    aws ec2 wait instance-exists \
```

Comment:
Not 100% sure if I'm reading the correct docs, but it seems to suggest that this operation polls by itself, and will return after 40 attempts (5 seconds wait between attempts): https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/wait/instance-exists.html

Would that mean we don't need our own polling logic here?

Reply:
You're definitely reading the correct docs, but we need to wait even longer than those 40 attempts for the build to finish...meaning we have to wrap the polling logic in polling logic 😅